mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -717,6 +719,51 @@ class SparseVector(
       new SparseVector(size, ii, vv)
     }
   }
+
+  override def argmax: Int = {
+    if (size == 0) {
+      -1
+    } else {
+
+      var maxIdx = indices(0)
+      var maxValue = values(0)
+
+      foreachActive { (i, v) =>
+        if (v > maxValue) {
+          maxIdx = i
+          maxValue = v
+        }
+      }
+
+      // look for inactive values in case all active node values are negative
+      if (size != values.size && maxValue <= 0) {
```

Comment:
We can implement this way:

~~~scala
var k = 0
while (k < indices.length && indices(k) == k) {
  k += 1
}
maxIdx = if (maxValue == 0.0 && k > maxIdx) maxIdx else k
~~~